pkg/config/remote/service/service.go

```diff
@@ -276,10 +276,33 @@ func (s *Service) ClientGetConfigs(request *pbgo.ClientGetConfigsRequest) (*pbgo
 	if err != nil {
 		return nil, err
 	}
+
+	directorTargets, err := s.uptane.Targets()
+	if err != nil {
+		return nil, err
+	}
+	configPointers, err := executeClientPredicates(request.Client, directorTargets)
+	if err != nil {
+		return nil, err
+	}
+
+	// filter files to only return the configPointers ones
+	matchedConfigsMap := make(map[string]interface{})
+	for _, configPointer := range configPointers {
+		matchedConfigsMap[configPointer] = struct{}{}
+	}
+	filteredFiles := make([]*pbgo.File, 0, len(configPointers))
+	for _, targetFile := range targetFiles {
+		if _, ok := matchedConfigsMap[targetFile.Path]; ok {
+			filteredFiles = append(filteredFiles, targetFile)
+		}
+	}
+
 	return &pbgo.ClientGetConfigsResponse{
-		Roots:       roots,
-		Targets:     targetsRaw,
-		TargetFiles: targetFiles,
+		Roots:         roots,
+		Targets:       targetsRaw,
+		TargetFiles:   filteredFiles,
+		ClientConfigs: configPointers,
```

Comment:
I think with my comment nits that it is clearer to new readers what's going on with this field, but this still reads (without context) like this would be the actual configs and not just a list of matched paths based on the predicates. Maybe `ClientConfigPaths` or `MatchedConfigPaths`?